seq2seq/metrics/metrics.py

```diff
@@ -356,4 +356,297 @@ def eval_batch(self, outputs, targets):
 
             # Check whether the prediction actually comes from the grammar
             if self.correct(grammar, prediction_correct_length):
-                self.seq_correct += 1
\ No newline at end of file
+                self.seq_correct += 1
+
+
+class VerifyProduceAccuracy(Metric):
+    """
+    Batch average of SciL produce micortask task sequence accuracy.
+    This metric is very specific for the produce micro task
+    For one input, multiple outputs can be correct.
+
+    Args:
+        input_vocab (torchtext.vocab): Input dictionary
+        output_vocab (torchtext.vocab): Output dictionary
+        use_output_eos (bool): Boolean to indicate whether an output EOS should be present
+        input_pad_symbol (str): Input PAD symbol
+        output_sos_symbol (str): Output SOS symbol
+        output_pad_symbol (str): Output PAD symbol
+        output_eos_symbol (str): Output EOS symbol
+        output_unk_symbol (str): Output UNK symbol
+    """
+
+    _NAME = "Microtask Accuracy"
+    _SHORTNAME = "vp_tsk_acc"
+    _INPUT = "seqlist"
+
+    def __init__(self, input_vocab, output_vocab, use_output_eos, input_pad_symbol, output_sos_symbol, output_pad_symbol, output_eos_symbol, output_unk_symbol):
+        self.input_vocab = input_vocab
+        self.output_vocab = output_vocab
+
+        self.use_output_eos = use_output_eos
+
+        # instead of passing all these arguments, we could also hard-code to use <sos>, <pad>, <unk> and <eos>
+        self.input_pad_symbol = input_pad_symbol
+        self.output_sos_symbol = output_sos_symbol
+        self.output_pad_symbol = output_pad_symbol
+        self.output_eos_symbol = output_eos_symbol
+        self.output_unk_symbol = output_unk_symbol
+
+        self.seq_correct = 0
+        self.seq_total = 0
+
+        super(VerifyProduceAccuracy, self).__init__(self._NAME, self._SHORTNAME, self._INPUT)
+
+    def get_val(self):
+        """
+        Get the average accuracy metric of all processed batches
+        Returns:
+            float: average accuracy
+        """
+        if self.seq_total != 0:
+            return float(self.seq_correct) / self.seq_total
+        else:
+            return 0
+
+    def reset(self):
+        """
+        Reset after all batches have been processed
+        """
+        self.seq_correct = 0
+        self.seq_total = 0
+
+    def correct(self, grammar, prediction):
+        '''
+        Return True if the target is a valid output given the source
+        Args:
+            grammar (list(str)): List of symbols of the grammar
+            prediction (list(str)): List of symbols of the prediction
+        Returns:
+            bool: whether the prediction is coming from the grammar
+        '''
+        all_correct = False
```

Comment:
Python does not require initialization of variables. This variable is set in all if-else clauses.